### PR TITLE
docs: fix link to redirects & rewrites, minor edits

### PR DESCRIPTION
**Pull request by rstavchansky**
### Summary
Fixes broken link from README to redirects & rewrites docs content in this repo.

### Test plan

1. Check out the diff, the [README on this branch](https://github.com/netlify/netlify-plugin-nextjs/blob/rms/docsLinkFix/README.md), and the [Redirects & rewrites page on this branch](https://github.com/netlify/netlify-plugin-nextjs/blob/rms/docsLinkFix/docs/redirects-rewrites.md) for edits in context.
   ...

### Relevant links (GitHub issues, Notion docs, etc.) or a picture of cute animal (small picture)
Late breaking review of https://github.com/netlify/netlify-plugin-nextjs/pull/767

---

### CODE REVIEW

File: docs/redirects-rewrites.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
...
+Every site on Netlify supports [redirects and rewrites](https://docs.netlify.com/routing/redirects/), which are defined in a `_redirects` file or `netlify.toml`, and sites that use this plugin are no exceptions. However there are some caveats to bear in mind when using them. The plugin generates several rewrites of its own, which are used to map paths from the site to different Netlify functions which handle SSR, preview mode and images, as well as assets in `/_next/static`. Any Netlify redirects or rewrites that you create [take precedence over these rewrites](#Redirect-and-rewrite-precedence), so you should avoid adding a root-level Netlify rewrite, because that would override the rewrites generated by and required by the plugin.
```
Comment (rstavchansky) [inline docs/redirects-rewrites.md:5]: It may be useful to add an example of what you mean by a root-level Netlify rewrite.

Comment (rstavchansky) [inline docs/redirects-rewrites.md:5]: It may be useful to add an example of what you mean by a root-level Netlify rewrite.

File: README.md [hunk 2]
```diff
@@ -42,7 +42,7 @@ If you are using a monorepo you will need to change `publish` to point to the fu
 
 If you are using Nx, then you will need to point `publish` to the folder inside `dist`, e.g. `dist/apps/myapp/.next`. 
 
-If you currently use redirects or rewrites on your site, see [the Rewrites and Redirects guide](https://github.com/netlify/netlify-plugin-nextjs/blob/main/docs/rewrites-and-redirects.md) for information on changes to how they are handled in this version.
+If you currently use redirects or rewrites on your site, see [the Rewrites and Redirects guide](https://github.com/netlify/netlify-plugin-nextjs/blob/main/docs/redirects-rewrites.md) for information on changes to how they are handled in this version.
```
Comment (rstavchansky) [inline README.md:45]: Link fix

File: docs/redirects-rewrites.md (hunk 1 above)
Comment (ascorbic) [reply docs/redirects-rewrites.md:5]: Good point

**Review by ascorbic — APPROVED** [review]

**Comment by ascorbic** [comment]
I'll add the root level rewrite in a follow-up

**Pull request merged**